### PR TITLE
Feature model

### PR DESCRIPTION
**Pull request by luigi-borriello00**
In this PR I would like to merge the following features:
- Reactive SetUp View, changing number of textbox according to the user preferencies
- A first Model Implementation, composed by:

1. Resource Cards Management
2. Development Cards Managment
3. Buildings Management
4. Awards Managment
5. Scores Managment

### CODE REVIEW

File: src/main/scala/scatan/model/game/Game.scala [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      result && resourceCards(player).count(_.resourceType == resourceCost._1) >= resourceCost._2
+    )
+
+  override def assignBuilding(building: Building, player: Player): Game =
+    this.copy(
+      buildings = buildings.updated(player, buildings(player) :+ building),
+      assignedAwards = awards
+    )
+
+  override def build(building: Building, player: Player): Game =
```
Comment (luigi-borriello00) [inline src/main/scala/scatan/model/game/Game.scala:97]: This method will be use when a player wants to Build something paying with his resources

File: src/main/scala/scatan/model/game/Game.scala [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    if verifyResourceCost(player, building.buildingType.cost) then
+      val remainingResourceCards = building.buildingType.cost.foldLeft(resourceCards(player))((cards, resourceCost) =>
+        cards.filter(_.resourceType != resourceCost._1).drop(resourceCost._2)
+      )
+      val gameWithConsumedResources = this.copy(resourceCards = resourceCards.updated(player, remainingResourceCards))
+      gameWithConsumedResources.assignBuilding(building, player)
+    else this
```
Comment (luigi-borriello00) [inline src/main/scala/scatan/model/game/Game.scala:104]: The difference between this two methods is that the `assignBuilding` doesn't check the actual resource avaiability necessary to build the construction, it's useful for the INITIAL PLACEMENT PHASE and it semplify Unit Testing.

File: src/main/scala/scatan/views/game/SetUpView.scala [hunk 3]
```diff
@@ -29,7 +32,28 @@ private class ScalaJsSetUpView(container: String, requirements: View.Requirement
...
     with SetUpView:
 
-  val numberOfUsers: Int = 3
+  val numberOfUsers: Var[Int] = Var(3)
+  val reactiveNumberOfUsers: Signal[Int] = numberOfUsers.signal
+
+  private def validateNames(usernames: String*) =
+    usernames.forall(_.matches(".*\\S.*"))
+
+  @SuppressWarnings(Array("org.wartremover.warts.AsInstanceOf"))
```
Comment (alemazzo) [inline src/main/scala/scatan/views/game/SetUpView.scala]: This should be removed

**Review by alemazzo — CHANGES_REQUESTED** [review]

**Review by alemazzo — APPROVED** [review]

**Review by alemazzo — APPROVED** [review]
Nice!

**Review by manuandru — APPROVED** [review]

**Pull request merged**